kubernetes/structures_deployment.go

```diff
@@ -90,40 +114,37 @@ func expandPodTemplate(l []interface{}) (*corev1.PodTemplateSpec, error) {
 }
 
 func expandDeploymentStrategy(l []interface{}) appsv1.DeploymentStrategy {
+	obj := appsv1.DeploymentStrategy{}
+
 	if len(l) == 0 || l[0] == nil {
-		return appsv1.DeploymentStrategy{}
+		obj.Type = appsv1.RollingUpdateDeploymentStrategyType
+		return obj
 	}
 	in := l[0].(map[string]interface{})
-	obj := appsv1.DeploymentStrategy{}
-	if v, ok := in["type"].(map[string]interface{}); ok && len(v) > 0 {
-		obj.Type = appsv1.DeploymentStrategyType(in["type"].(string))
+
+	if v, ok := in["type"]; ok {
+		obj.Type = appsv1.DeploymentStrategyType(v.(string))
 	}
-	if v, ok := in["rolling_update"].([]interface{}); ok && len(v) > 0 {
-		obj.RollingUpdate = expandRollingUpdateDeployment(v)
+	if v, ok := in["rolling_update"]; ok {
```

Comment:
Probably we should check `in["type"]` if it is `string` and length of `v`
Like
```
if v, ok := in["rolling_update"].([]interface{}); ok && len(v) > 0 {
```

Reply:
I don't think there is any practical benefit to doing this check because `in["rolling_update"]` will either be empty or a slice of length 1 per the schema. Am I missing something?

Reply:
You are right, my bad. Thank you for figuring out.
Only reason I see is to be consistent with other resources.

Reply:
I ended up needing this for the scenario where `rolling_update` is missing and `type` is specified. Turns out there is a practical benefit :)! Thanks for the suggestion